packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -0,0 +1,470 @@
+                    this.incrementSelectedOption(+1);
+                }
+                break;
+            case KeyCodes.arrowUp:
+            case KeyCodes.arrowLeft:
+                if (!this.props.multiselectable && !e.defaultPrevented) {
+                    this.incrementSelectedOption(-1);
+                }
+                break;
+        }
+    };
+
+    /**
+     * Increment selection
+     */
+    private incrementSelectedOption = (increment: number): void => {
+        const childrenAsArray: React.ReactNode[] = React.Children.toArray(
+            this.props.children
+        );
+        const iterationIncrement: number = increment > -1 ? 1 : -1;
+        if (this.state.selectedItems.length === 1) {
+            const selectedItemIndex: number = Listbox.getItemIndexById(
+                this.state.selectedItems[0].id,
+                this.props.children
+            );
+            if (selectedItemIndex !== -1) {
+                const startIndex: number = selectedItemIndex + iterationIncrement;
+                const endIndex: number = increment > -1 ? 0 : childrenAsArray.length - 1;
+                if (startIndex >= 0 && startIndex < childrenAsArray.length) {
+                    const validOption: React.ReactNode = Listbox.getFirstValidOptionInRange(
+                        startIndex,
+                        endIndex,
+                        childrenAsArray,
+                        increment
+                    );
+                    if (validOption !== null) {
+                        this.updateSelection([
+                            (validOption as React.ReactElement<any>).props,
+                        ]);
+                    }
+                }
+            }
+        } else {
+            const startIndex: number = increment > -1 ? 0 : childrenAsArray.length - 1;
+            const endIndex: number = increment > -1 ? childrenAsArray.length - 1 : 0;
+            const validOption: React.ReactNode = Listbox.getFirstValidOptionInRange(
+                startIndex,
+                endIndex,
+                childrenAsArray,
+                increment
+            );
+            if (validOption !== null) {
+                this.updateSelection([(validOption as React.ReactElement<any>).props]);
+            }
+        }
+
+        this.toggleMenu(true);
+        return;
+    };
+
+    /**
+     * Toggles the menu
+     */
+    private toggleMenu = (isMenuOpen: boolean): void => {
+        let shouldOpenMenu: boolean = isMenuOpen;
+        if (this.props.isMenuOpen !== undefined) {
+            shouldOpenMenu = this.props.isMenuOpen;
+        } else if (this.props.multiselectable === true) {
+            shouldOpenMenu = true;
+        } else {
+            if (shouldOpenMenu) {
+                window.addEventListener("click", this.handleWindowClick);
+            } else {
+                window.removeEventListener("click", this.handleWindowClick);
+            }
+        }
```

Comment:
Do we want to be adding and removing listeners for this over and over? That seems like something we're over-managing. Why not just add this when the component mounts and remove when the component unmounts?

Reply:
Agreed.  Changed it.